src/invariant/LiabilitiesMatchOffers.cpp

```diff
@@ -87,7 +88,7 @@ checkAuthorized(std::shared_ptr<LedgerEntry const> const& current)
     if (current->data.type() == TRUSTLINE)
     {
         auto const& trust = current->data.trustLine();
-        if (!(trust.flags & AUTHORIZED_FLAG))
+        if (!isAuthorizedToMaintainLiabilities(*current))
```

Comment:
Liabilities can still increase when `AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG` is set. When an offer with an asset in the maintain liabilities state get's filled, liabilities can increase.

Reply:
That doesn't sound right to me. Liabilities should only decrease when an offer executes, because the liabilities are themselves a bound on how much your balance might change if an offer were to execute.

Reply:
Sorry, my statement wasn't clean. The scenario I was thinking about is when the offer in the book is partially filled. That order initially has its liabilities released. After the partial fill, that order needs to be put back into the book, which will increase liabilities. 

Does that sounds right?

Reply:
That's correct. But the invariants are only checked after every operation, never in the middle of an operation, so it is still safe to check for this.

Reply:
Good point. Added another invariant in `checkAuthorized`.